services/httpd/handler.go

```diff
@@ -993,8 +1012,366 @@ func fixLiterals(node influxql.Node) {
 	}
 }
 
+type RetentionRule struct {
+	Type                      string `json:"type"`
+	EverySeconds              int64  `json:"everySeconds"`
+	ShardGroupDurationSeconds int64  `json:"shardGroupDurationSeconds"`
+}
+
+// BucketsBody and RetentionRule should match the 2.0 API definition.
+type BucketsBody struct {
+	Description    string          `json:"description"`
+	Name           string          `json:"name"`
+	OrgID          string          `json:"orgId"`
+	Rp             string          `json:"rp"`
```

Comment:
In v2 you can give an `Rp` in the `POST` request, but  I can't see how that would be useful and you don't get it back in the `GET` request. Maybe it's just dead and should be removed?

Reply:
You do get it back in the GET request: https://docs.influxdata.com/influxdb/cloud/api/#operation/GetBucketsID

I think it should just be the retention policy name, so it should match the parsed-out RP from the bucket name on POST, and be provided on the GET from the bucket name.

Reply:
We check for a match on POST in line. rp has to match or be empty.